v3: GRPC client invoke method builder panics for empty messages

Here's a reproducible design:

```go
package design

import . "goa.design/goa/v3/dsl"

var _ = API("dummy", func() {
	Title("Dummy Service")
	Description("service for testing purposes")
	Server("dummy", func() {
		Host("localhost", func() { URI("http://localhost:8000") })
	})
})

var _ = Service("dummy", func() {
	Method("doStuff", func() {
		Description("Do stuff endpoint")

		Result(func() {
			Field(1, "message", String, "Details about stuff done")
			Required("message")
		})

		GRPC(func() {
			Response(CodeOK)
			Response(CodeUnavailable)
		})
	})
})
```
And then running the following commands in sequence:
```
goa gen dummy/design
```
```
goa example dummy/design
```
```
go build ./cmd/dummy && go build ./cmd/dummy-cli
```
Running the server: `./dummy`
And then making a client call: `./dummy-cli dummy do-stuff`

## Expected output
```
{
  "Message": ""
}
```

## Actual output
```
panic: interface conversion: interface {} is nil, not *dummypb.DoStuffRequest

goroutine 1 [running]:
test/gen/grpc/dummy/client.BuildDoStuffFunc.func1(0x965140, 0xc000077e60, 0x0, 0x0, 0xc000004b60, 0x2, 0x2, 0xc000077e60, 0x0, 0xc79a60, ...)
        E:/test/gen/grpc/dummy/client/encode_decode.go:26 +0x234
goa.design/goa/v3/grpc.(*cliInvoker).Invoke(0xc000004b20, 0x9650c0, 0xc000018110, 0x0, 0x0, 0xc000077e00, 0xc000056060, 0x8d0312, 0x0)
        E:/go/pkg/mod/goa.design/goa/v3@v3.0.2/grpc/client.go:66 +0x27b
test/gen/grpc/dummy/client.(*Client).DoStuff.func1(0x9650c0, 0xc000018110, 0x0, 0x0, 0x1e, 0x0, 0xc000055760, 0x0)
        E:/test/gen/grpc/dummy/client/client.go:40 +0xde
main.main()
        E:/test/cmd/dummy-cli/main.go:82 +0x542
```

This is the referenced source of the panic (`./gen/grpc/dummy/client/encode_decode.go`):
```go
// BuildDoStuffFunc builds the remote method to invoke for "dummy" service
// "doStuff" endpoint.
func BuildDoStuffFunc(grpccli dummypb.DummyClient, cliopts ...grpc.CallOption) goagrpc.RemoteFunc {
	return func(ctx context.Context, reqpb interface{}, opts ...grpc.CallOption) (interface{}, error) {
		for _, opt := range cliopts {
			opts = append(opts, opt)
		}
		return grpccli.DoStuff(ctx, reqpb.(*dummypb.DoStuffRequest), opts...)
	}
}
```

--- a/grpc/codegen/client.go
+++ b/grpc/codegen/client.go
@@ -240,7 +240,10 @@ func Build{{ .Method.VarName }}Func(grpccli {{ .PkgName }}.{{ .ClientInterface }
 		for _, opt := range cliopts {
 			opts = append(opts, opt)
 		}
-		return grpccli.{{ .Method.VarName }}(ctx{{ if not .Method.StreamingPayload }}, reqpb.({{ .Request.ClientConvert.TgtRef }}){{ end }}, opts...)
+		if reqpb != nil {
+			return grpccli.{{ .Method.VarName }}(ctx{{ if not .Method.StreamingPayload }}, reqpb.({{ .Request.ClientConvert.TgtRef }}){{ end }}, opts...)
+		}
+		return grpccli.{{ .Method.VarName }}(ctx{{ if not .Method.StreamingPayload }}, &{{ .Request.ClientConvert.TgtName }}{}{{ end }}, opts...)
 	}
 }
 `

